.github/workflows/build_charm.yaml

```diff
@@ -143,12 +143,12 @@ jobs:
         run: |
           if '${{ inputs.cache }}'
           then
-            sg lxd -c "charmcraftcache pack -v --bases-index='${{ matrix.base.id }}'"
+            sudo --user "$USER" --preserve-env --preserve-env=PATH -- env -- charmcraftcache pack -v --bases-index='${{ matrix.base.id }}'
```

Comment:
This doesn't seem to be enough. Testing on a [branch](https://github.com/canonical/postgresql-k8s-operator/actions/runs/12632304086/job/35196633551?pr=808) failed with `sudo: charmcraftcache: command not found` prepending `pipx ensurepath` before calling ccc fails with ```LXD requires additional permissions. Recommended resolution: Ensure that the user is in the 'lxd' group.```

Reply:
Updated the branch and the packing passed.